File: Backend/StockDaddy.Application/DTOs/CustomerDto.cs
namespace StockDaddy.Application.DTOs;

public class CustomerDto
{
    public Guid Id { get; set; }
    public Guid TenantId { get; set; }

    public string Name { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public bool IsDeleted { get; set; }
    public DateTime? DeletedAt { get; set; }
}

File: Frontend/src/dtos/customer.dto.ts
export interface CustomerDto {
  id: number;
  tenantId: number;
  name: string;
  phone: string;
  email: string;
  address: string;
  createdAt: string;
  updatedAt: string;
  isDeleted: boolean;
}

export interface CreateCustomerRequest {
  tenantId: number;
  name: string;
  phone: string;
  email: string;
  address: string;
}

export interface UpdateCustomerRequest {
  name: string;
  phone: string;
  email: string;
  address: string;
}
